www_src/lib/cartesian.js

```diff
@@ -0,0 +1,129 @@
+class Cartesian {
+
+  constructor(options) {
+    options = options || {};
+    if (!options.height || !options.width || !options.allCoords) {
+      throw new Error('You must configure height, width, allCoords');
+    }
+    this.width = options.width;
+    this.height = options.height;
+    this.gutter = options.gutter || 0;
+    this.allCoords = options.allCoords;
+  }
+
+  get edges() {
+    return this.getEdges();
+  }
+
+  // Includes edges.
+  get all() {
+    return this.allCoords.concat(this.edges);
+  }
+
+  get allX() {
+    return this.all.map(xy => xy.x);
+  }
+  get allY() {
+    return this.all.map(xy => xy.y);
+  }
+  get widthPadded() {
+    return this.width + this.gutter;
+  }
+  get heightPadded() {
+    return this.height + this.gutter;
+  }
+
+  // Calculate absolute distance from 0 to min/max
+  getAbsDistance(xOrY, maxOrMin) {
```

Comment:
I'm having a hard time grokking this function, but could be my geometry being rusty...